shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -79,12 +79,24 @@ class StepQuizActionDispatcher(
                         return
                     }
 
+                val isProblemsLimitReached = freemiumInteractor
+                    .isProblemsLimitReached()
```

Comment:
Is it loading from remote source or from in-memory, if from remote we should parallel it

Reply:
To the moment of opening Step Quiz screen subscription will be loaded to memory, so no need to optimize